### PR TITLE
Prepare alloc library for paging system

### LINKED ISSUE #9: Prepare alloc library for paging system
### Summary
Prepare source and header files for issues #7 and #8 

### Tasks
- [x] Remove old `heapalloc.c`
- [x] Remove `alloc.h`
- [x] Rename `globalalloc.c` to `heapalloc.c`
- [x] Create `heapalloc.h`
- [x] Add `struct heap_state` to `heapalloc.h`
- [x] Change `heapalloc.h` to conform to new specification
- [x] Modify linker script to add new memory section for pages
- [x] Create `pagealloc.c` and `pagealloc.h`
- [x] Modify `stdalloc.c` to conform to new `heapalloc` specification

### PR DESCRIPTION
**Pull request by jakejack13**

### CODE REVIEW

File: src/stdlib/stdalloc.c [hunk 1]
```diff
@@ -1,25 +1,25 @@
 #include "stdlib/stdalloc.h"
 
-#include "alloc/alloc.h"
-#include "config.h"
+#include "alloc/heapalloc.h"
 
 /** The currently used standard allocator */
```
Comment (jakejack13) [inline src/stdlib/stdalloc.c]: The line of documentation is not up to date

**Review by jakejack13 — COMMENTED** [review]
Slight issue with documentation

**Review by warrenyun — APPROVED** [review]
looks good!

**Review by anglesideangle — APPROVED** [review]
good!

**Pull request merged**